### PR TITLE
Unified RAG Blog Post and Guide

### PR DESCRIPTION
**Pull request by SilasMarvin**

### CODE REVIEW

File: pgml-cms/docs/guides/unified-rag.md [hunk 1]
```diff
@@ -0,0 +1,332 @@
+---
+description: >-
+  Unified RAG is an alternative to typically RAG systems where embedding, retrieval, reranking, and text-generation are combined under on service.
+---
```
Comment (montanalow) [inline pgml-cms/docs/guides/unified-rag.md:4]:
```suggestion
featured: true
---
```

File: pgml-cms/docs/guides/unified-rag.md [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+# Unified RAG
+
+## Introduction <a href="#introduction" id="introduction"></a>
```
Comment (montanalow) [inline pgml-cms/docs/guides/unified-rag.md:8]: Our cms should add the heading anchors for you automatically
```suggestion
## Introduction
```

File: pgml-cms/docs/guides/unified-rag.md [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+
+RAG has grown rapidly in popularity. It is not an esoteric practice run only by advanced machine learning practitioners, but is used widely by anyone who wants to improve the output of their LLMs. It is most commonly used by chatbots to better answer user questions.
+
+As quick reminder, the typical modern RAG workflow looks like this:
+
+TODO: INSERT RAG DIAGRAM
+
+Steps one through three prepare our RAG system, and steps four through eight are RAG itself. 
+
+This is not a post about the typically modern RAG workflow, this is a demonstration of Unified RAG and the simplicity and power it provides.
```
Comment (montanalow) [inline pgml-cms/docs/guides/unified-rag.md:22]: This feels like the lede is buried.

File: pgml-cms/docs/guides/unified-rag.md [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+
+Notice that we set expanded display to auto to make it easier to visualize the output.
+
+### Unified Retrieval
+
+Retrieval with Unified RAG is lightning fast and incredibly simple. 
+
+!!! generic
+
+!!! code\_block time="70.381 ms"
```
Comment (montanalow) [inline pgml-cms/docs/guides/unified-rag.md:240]: These underscores don't need to be escaped
```suggestion
!!! code_block time="70.381 ms"
```

**Review by montanalow — COMMENTED** [review]
Looking good so far!

File: pgml-cms/docs/guides/unified-rag.md [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+    |                     |     "max_new_tokens": 100                                   +
+    |                     |   }'::JSONB                                                 +
+    |                     | );
+(6 rows)
+
+!!!
+
+!!!
+
+We are using a CTE to embed the user query, and then performing nearest neighbors search using the cosine similarity function to compare the distance between our embeddings. Note how fast this is! Our embeddings utilize an HNSW index from pgvector to perform ridiculously fast retrieval.
```
Comment (katopz) [inline pgml-cms/docs/guides/unified-rag.md:310]: What's `CTE` stand for? 🤔

Comment (SilasMarvin) [reply pgml-cms/docs/guides/unified-rag.md:311]: It stands for common table expression: https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-cte/

**Comment by montanalow** [comment]
🚀

**Pull request merged**